code/weapon/weapons.cpp

```diff
@@ -7770,6 +7818,9 @@ void weapon_info::reset()
 	this->thruster_glow_factor = 1.0f;
 	this->target_lead_scaler = 0.0f;
 
+	this->failure_rate = 0.0f;
```

Comment:
This block does not initialize `failure_sub_name` which is probably needed here. Even if it is not strictly needed it is better to explicitly reset all fields.